### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'get_type_string'

### DISCUSSION THREAD

**Discussion by klemendrobnic** [polkascan/py-substrate-interface#376] [Q&A] [answered]
I would need help with resolving an error that occurs when im trying to use compose_call function.

```
from substrateinterface import SubstrateInterface

substrate = SubstrateInterface(url="wss://ws.framenode-7.s4.stg1.sora2.soramitsu.co.jp")


call = substrate.compose_call(call_module='Balances',call_function='transfer',call_params={'dest': '5CvxUr1gXDvAN8qEDPBFVdnF7QcPwNfYCQLYDDr3cCzC45Vt','value': 1 * 10**12})
```

I get the following error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/klemendrobnic/.local/lib/python3.10/site-packages/substrateinterface/base.py", line 1408, in compose_call
    call.encode({
  File "/home/klemendrobnic/.local/lib/python3.10/site-packages/scalecodec/base.py", line 929, in encode
    self.data = self.process_encode(self.value_serialized)
  File "/home/klemendrobnic/.local/lib/python3.10/site-packages/scalecodec/types.py", line 1652, in process_encode
    call_type_string = self.call_module['calls'].value_object.get_type_string()
AttributeError: 'NoneType' object has no attribute 'get_type_string'
```
When i print the call_module that is called in the process_encode method i get the object printed, however the parameter calls is not included (call_type_string = self.call_module['calls'].value_object.get_type_string()) -> the code fails here
Thanks for any suggestions

**Comment by arjanz** [comment, marked as answer]
After examining the metadata of the runtime, I found out that the `Balances` pallet for the endpoint you provided has in fact no runtime calls at all (so call `transfer` couldn't be found). You can also check [PolkadotJS](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fws.framenode-7.s4.stg1.sora2.soramitsu.co.jp#/extrinsics)

However, the error handler could be improved here, the current thrown exception is indeed confusing and should be something like `Call function 'transfer' not found`. I'll make an issue for this.

**Reply by klemendrobnic** [reply]
The solution here was that the call_module=Balances, therefore the Balances pallet doesn't exist on the node that I was trying to connect to - Sora network. The solution was to replace the call_module to call_module=Assets and now it works. Thank you for your reply.
